### PR TITLE
UF Docks implementation

### PR DESCRIPTION
**Pull request by ederign**
Requires : https://github.com/uberfire/uberfire/pull/144

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -46,6 +46,7 @@
     <module>uberfire-wires</module>
     <module>uberfire-apps</module>
     <module>uberfire-layout-editor</module>
+    <module>uberfire-docks</module>
```
Comment (porcelli) [inline pom.xml]: what about uberfire-simple-docks ?

File: uberfire-docks/uberfire-docks-client/src/main/java/org/uberfire/client/docks/DocksBar.java [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        }
+    }
+
+    public void deselectAllDocks() {
+        for ( DocksItem docksItem : docksItems ) {
+            docksItem.deselect();
+        }
+    }
+
+    public void addDock( UberfireDock dock,
```
Comment (porcelli) [inline uberfire-docks/uberfire-docks-client/src/main/java/org/uberfire/client/docks/DocksBar.java]: why not final too?

File: uberfire-wires/uberfire-wires-webapp/.gitignore [hunk 3]
```diff
@@ -2,7 +2,7 @@
 /src/main/webapp/WEB-INF/classes/
 /src/main/webapp/WEB-INF/deploy/
 /src/main/webapp/WEB-INF/lib/
-
+/src/main/webapp/org.uberfire.ext.wires.WiresShowcase/
```
Comment (ederign) [inline uberfire-wires/uberfire-wires-webapp/.gitignore]: @porcelli

**Comment by porcelli** [comment]
https://github.com/uberfire/uberfire-extensions/commit/90a0009eb

**Pull request closed without merging**